sklearn/utils/extmath.py

```diff
@@ -211,28 +219,39 @@ def randomized_range_finder(A, size, n_iter,
         else:
             power_iteration_normalizer = 'LU'
 
+    if subtract_mean:
+        c = A.mean(axis=0).reshape((1, -1))
+        applyA = lambda X: safe_sparse_dot(A, X) - safe_sparse_dot(c, X)
+        applyAT = lambda X: safe_sparse_dot(A.T, X) - \
+                            safe_sparse_dot(c.T, Q.sum(axis=0).reshape((1, -1)))
+    else:
+        applyA = lambda X: safe_sparse_dot(A, X)
+        applyAT = lambda X: safe_sparse_dot(A.T, X)
+
+    Q = applyA(Q)
+
```

Comment:
So when we take the QR or LU decomposition of A, it's just another representation of A. We keep only the Q and L matrices because they're the ones that contain the basis of A, and that's really all we care about.

Changing the order has no effect on the maths, but I'll change the order back to how it was before.